modules/stanford_profile_styles/stanford_profile_styles.module

```diff
@@ -34,18 +33,57 @@ function stanford_profile_styles_page_attachments(array &$attachments) {
     return;
   }
 
+  $node_type = $node->getType();
+  $attachments['#attached']['library'][] = "stanford_profile_styles/node.$node_type";
+
   // Check for our type and add library if a match.
-  if ($node->getType() == "stanford_page") {
-    $attachments['#attached']['library'][] = 'stanford_profile_styles/stanford_page';
+  if ($node_type == "stanford_page") {
 
     // Check if stanford page is using a particular layout.
     $layout_target = $node->get('layout_selection')->getValue();
     if (isset($layout_target[0]['target_id']) && $layout_target[0]['target_id'] == "stanford_basic_page_full") {
-      $attachments['#attached']['library'][] = "stanford_profile_styles/stanford_page_layout_full";
+      $attachments['#attached']['library'][] = "stanford_profile_styles/node.stanford_page.layout.full-width";
     }
   }
 }
 
+/**
+ * Implements hook_library_info_build().
+ */
+function stanford_profile_styles_library_info_build() {
+  $libraries = [];
+  $module_path = drupal_get_path('module', 'stanford_profile_styles');
+
+  // Find all css files in the dist/css directory.
+  $finder = new Finder();
+  $finder->in("$module_path/dist/css")
+    ->files()
+    ->name('/.css$/');
+
+  foreach ($finder->getIterator() as $file) {
+    $local_path = str_replace("$module_path/", '', $file->getPath());
+
+    $path_parts = explode('/', $local_path);
+    // Remove `dist` and `css` parts.
+    unset($path_parts[0], $path_parts[1]);
+
+    // This is the directory the file lives in.
+    $bucket = reset($path_parts);
+    $lib = basename($file->getFilename(), '.css');
+
+    // Build the library definition.
+    $libraries[trim("$bucket.$lib", '. ')] = [
+      'css' => [
+        'component' => [
+          "$local_path/{$file->getFileName()}" => [],
+        ],
```

Comment:
This is mighty slick but mighty assumptive. Component might not always be the best option for each css file. Perhaps you can come up with a naming convention that will allow for stylesheets to be placed at different levels? eg: `thing.base.css` or `thing.theme.css` and default to component if not specified.

Reply:
hmm. that's a pretty neat idea. i struggled with trying to figure that out. i like that idea.

Reply:
I created level directories to create these. should be pretty straight forward now.